### PR TITLE
More mbedtls_x509_crt_parse_path() tests, and note qemu-user bug when 32-bit code run on 64-bit host

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**
Document a [known issue](https://gitlab.com/qemu-project/qemu/-/issues/263) when running tests under Qemu (which we intend to do on the CI [one of these days](https://github.com/Mbed-TLS/mbedtls/issues/3855)).

## Gatekeeper checklist

- [x] **changelog** N/A
- [x] **backport** https://github.com/Mbed-TLS/mbedtls/pull/7483
- [x] **tests** N/A

### CODE REVIEW

File: tests/suites/test_suite_x509parse.data [hunk 1]
```diff
@@ -2432,15 +2432,29 @@ X509 CRL ASN1 (extension not critical explicit, crl-idp.pem byte 129)
...
+
+X509 CRT parse file dir3/test-ca.crt
+depends_on:MBEDTLS_MD_CAN_SHA1:MBEDTLS_RSA_C
+mbedtls_x509_crt_parse_file:"data_files/dir3/test-ca.crt":0:1
+
+X509 CRT parse file dir3/test-ca2.crt
+depends_on:MBEDTLS_MD_CAN_SHA256:MBEDTLS_PK_CAN_ECDSA_SOME:MBEDTLS_ECP_DP_SECP384R1_ENABLED
+mbedtls_x509_crt_parse_file:"data_files/dir3/test-ca2.crt":0:1
+
+# The parse_path tests are known to fail when compiled for a 32-btt architecture
```
Comment (tom-cosgrove-arm) [inline tests/suites/test_suite_x509parse.data:2446]:
```suggestion
# The parse_path tests are known to fail when compiled for a 32-bit architecture
```

**Review by tom-cosgrove-arm — DISMISSED** [review]
LGTM

File: tests/suites/test_suite_x509parse.data [hunk 2]
```diff
@@ -2432,15 +2432,29 @@ X509 CRL ASN1 (extension not critical explicit, crl-idp.pem byte 129)
...
+# and run via qemu-user on Linux on a 64-bit host. This is due to a known
+# bug in Qemu: https://gitlab.com/qemu-project/qemu/-/issues/263
+X509 CRT parse path #1 (one cert)
 depends_on:MBEDTLS_MD_CAN_SHA1:MBEDTLS_RSA_C
 mbedtls_x509_crt_parse_path:"data_files/dir1":0:1
 
-X509 CRT parse path #3 (two certs)
-depends_on:MBEDTLS_MD_CAN_SHA1:MBEDTLS_RSA_C:MBEDTLS_MD_CAN_SHA256:MBEDTLS_PK_CAN_ECDSA_SOME:MBEDTLS_ECP_DP_SECP384R1_ENABLED
+X509 CRT parse path #2 (two certs)
+depends_ondepends_on:MBEDTLS_MD_CAN_SHA1:MBEDTLS_RSA_C:MBEDTLS_MD_CAN_SHA256:MBEDTLS_PK_CAN_ECDSA_SOME:MBEDTLS_ECP_DP_SECP384R1_ENABLED
```
Comment (daverodgman) [inline tests/suites/test_suite_x509parse.data:2454]: Typo?

**Review by daverodgman — CHANGES_REQUESTED** [review]

**Review by daverodgman — APPROVED** [review]

**Comment by gilles-peskine-arm** [comment]
I forced-pushed because there was a merge conflict. In "Improve testing of mbedtls_x509_crt_parse_file", reconcile the change to the function that tests `mbedtls_x509_crt_parse_file`: addition of `USE_PSA_INIT/USE_PSA_DONE` in development, additional testing and moved to be grouped with `mbedtls_x509_crt_parse_path` in the PR.

File: tests/suites/test_suite_x509parse.data (hunk 1 above)
Comment (tom-cosgrove-arm) [reply tests/suites/test_suite_x509parse.data]: @daverodgman if you're fixing the conflicts, can you take this suggestion please?

Comment (tom-cosgrove-arm) [reply tests/suites/test_suite_x509parse.data]: err, n/m, already done

**Comment by daverodgman** [comment]
Discussed in historical review - conflicts are trivial so agreed to resolve the merge & dust off the review

**Review by daverodgman — APPROVED** [review]
LGTM

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

**Comment by daverodgman** [comment]
OpenCI failure is spurious

**Pull request merged**